app/Http/Middleware/EnsureStatus.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\Models\Semester;
+use App\Models\SemesterStatus;
+
+use Closure;
+use Illuminate\Http\Request;
+
+class EnsureStatus
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        if (!$request->is('logout') && !$request->routeIs('secretariat.status-update.*') && $request->user() && $request->user()->getStatusIn(Semester::current())==SemesterStatus::INACTIVE) {
```

Comment:
add isCollegist()

Reply:
And maybe this could be simpler if you just add all these to the isStatementAvailable function

Reply:
You could add this to the routes as a withoutmiddleware for the logout.
Also, rename this file for a more general-purpose one where we can have more redirection rules in the future.

Reply:
The `withoutMiddleware` function didn't really work, I just forgot to delete it in the code. And I don't think that we should be adding this to the `isStatementAvailable `function as these are not checking if it's available.